### PR TITLE
Remove plugin search feature

### LINKED ISSUE #8056: [Bug?] Symlink'd plugins in `node_modules` are not auto detected.
**Environments:**

- Prettier Version: 2.0.4
- Running Prettier via: VSCode, CLI
- Runtime: Node 12.16.2
- Operating System: OS X

**Steps to reproduce:**

1. Use `npm` to install  [prettier-plugin-organize-imports](https://www.npmjs.com/package/prettier-plugin-organize-imports)
2. Run `prettier foo.txs`
3. Observe that imports are correctly organized

This time, use `pnpm` which uses symlinks instead of copying packages to `node_modules`:

1. Use `pnpm` to install [prettier-plugin-organize-imports](https://www.npmjs.com/package/prettier-plugin-organize-imports)
2. Run `prettier foo.tsx` from the command line
3. Observe that imports are *NOT* organized

Now, add the plugin explicitly to `package.json`:

```json
  "prettier": {
    "plugins": [
      "./node_modules/prettier-plugin-organize-imports"
    ]
  }
```

Repeat step 2 above and observe that plugins are now organized.

**Expected behavior:**

Symlink'd plugins in `node_modules` are auto detected.

**Actual behavior:**

Symlink'd plugins in `node_modules` are not auto detected.

### LINKED ISSUE #8474: Plugin autoload doesn't work for packages installed from GitHub and for Yarn PnP
**Environments:**

- Prettier Version: 2.0.5
- Running Prettier via: CLI
- Runtime: Node.js v14
- Operating System: Windows

**Steps to reproduce:**

```console
yarn add prettier/prettier @prettier/plugin-pug
```

**Expected behavior:**

Can format pug file with `npx prettier test.pug`

**Actual behavior:**

The CLI can't load `pug` plugin automatically

----

Problem should be [here](https://github.com/prettier/prettier/blob/6b4c4066aa1d59e6f806b1646beee314f93103d3/src/common/load-plugins.js#L30), when install from github, `./node_modules/prettier` maybe has own `node_modules` dir, so the `autoLoadDir` will be `./node_modules/prettier`, not expected `./`, so we are looking for `./node_modules/prettier/node_modules/@prettier/plugin-*` not `./node_modules/@prettier/plugin-*` (where the `@prettier/plugin-pug` is installed).

### LINKED ISSUE #14652: getSupportInfo() returns different results based on pnpm or npm
### Steps To Reproduce:
1. Create a simple repo following instructions here: https://kit.svelte.dev/docs/creating-a-project
2. After `npm install`:

```
Welcome to Node.js v18.15.0.
Type ".help" for more information.
> require('prettier').getSupportInfo()['languages'].map(e=>e.vscodeLanguageIds)
[
  [ 'javascript', 'mongo' ],
  [ 'javascript' ],
  [ 'javascriptreact' ],
  [ 'typescript' ],
  [ 'typescriptreact' ],
  [ 'json' ],
  [ 'json' ],
  [ 'jsonc' ],
  [ 'json5' ],
  [ 'css' ],
  [ 'postcss' ],
  [ 'less' ],
  [ 'scss' ],
  [ 'handlebars' ],
  [ 'graphql' ],
  [ 'markdown' ],
  [ 'mdx' ],
  [ 'html' ],
  [ 'html' ],
  [ 'html' ],
  [ 'vue' ],
  [ 'yaml', 'ansible', 'home-assistant' ],
  [ 'svelte' ]
]
```

3. After `rm -rf node_modules && pnpm install`:
```
Welcome to Node.js v18.15.0.
Type ".help" for more information.
> require('prettier').getSupportInfo()['languages'].map(e=>e.vscodeLanguageIds)
[
  [ 'javascript', 'mongo' ],
  [ 'javascript' ],
  [ 'javascriptreact' ],
  [ 'typescript' ],
  [ 'typescriptreact' ],
  [ 'json' ],
  [ 'json' ],
  [ 'jsonc' ],
  [ 'json5' ],
  [ 'css' ],
  [ 'postcss' ],
  [ 'less' ],
  [ 'scss' ],
  [ 'handlebars' ],
  [ 'graphql' ],
  [ 'markdown' ],
  [ 'mdx' ],
  [ 'html' ],
  [ 'html' ],
  [ 'html' ],
  [ 'vue' ],
  [ 'yaml', 'ansible', 'home-assistant' ]
]
```

Notice how `svelte` only appears with `npm`

Relevant .prettierrc config:
``` 
{
        "useTabs": true,
        "singleQuote": true,
        "trailingComma": "none",
        "printWidth": 100,
        "semi": false,
        "plugins": ["prettier-plugin-svelte"],
        "pluginSearchDirs": ["."],
        "overrides": [{ "files": "*.svelte", "options": { "parser": "svelte" } }]
}
```

### PR DESCRIPTION
**Pull request by fisker**
## Description

Closes #14652
Closes #8056
Closes #9167
Closes #13583
Closes #8474

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: docs/plugins.md [hunk 1]
```diff
@@ -35,18 +30,11 @@ When plugins cannot be found automatically, you can load them with:
...
   ```
 
-`pluginSearchDirs` and `plugins` are independent and one does not require the other.
-
-The paths that are provided to `pluginSearchDirs` will be searched for `@prettier/plugin-*`, `prettier-plugin-*`, and `@*/prettier-plugin-*`. For instance, these can be your project directory, a `node_modules` directory, the location of global npm modules, or any arbitrary directory that contains plugins.
-
-Strings provided to `plugins` are ultimately passed to `require()`, so you can provide a module/package name, a path, or anything else `require()` takes. (`pluginSearchDirs` works the same way. That is, valid plugin paths that it finds are passed to `require()`.)
-
-To turn off plugin autoloading, use `--no-plugin-search` when using Prettier CLI or add `{ pluginSearchDirs: false }` to options in `prettier.format()` or to the config file.
+Strings provided to `plugins` are ultimately passed to `import()`, so you can provide a module/package name, a path, or anything else `import()` takes.
```
Comment (sosukesuzuki) [inline docs/plugins.md:37]: Nits for Node.js beginners

```suggestion
Strings provided to `plugins` are ultimately passed to [`import()` expression](https://nodejs.org/api/esm.html#import-expressions), so you can provide a module/package name, a path, or anything else `import()` takes.
```

**Review by sosukesuzuki — APPROVED** [review]
I love changes like this that remove features as a maintainer, thank you. (One nits)

**Comment by fisker** [comment]
@kachkaev 

Reply to your question in discord about "global installation".

I don't have a good answer for that, they'll have to run or alias `prettier --plugin=prettier-plugin-foo` or `prettier --config=/path/to/global/.prettierrc`.

**Comment by kachkaev** [comment]
@fisker maybe there could be some solution offered on the editor extension level. I believe that global Prettier setup is mainly needed in somewhat non-standard environments like in education. For example, [prettier-plugin-elm](https://github.com/gicentre/prettier-plugin-elm) is used with [litvis](https://github.com/gicentre/litvis) for teaching data visualisation.

In general, I am supportive of plugin search removal – this feature is really hard to implement for all package managers and various modes within them. We just need to be aware of us breaking certain valid use cases and come up with a new solution for them. Perhaps, folks who do rely on global plugin search and can’t easily use `import()` in `prettier.config.js` can stick with v2 until a new solution is being established.

**Comment by vinkla** [comment]
Is there a possibility that this pull request could resolve the hoisting issue in pnpm?

https://github.com/pnpm/pnpm/issues/4700

**Comment by eldair** [comment]
Could this be the cause of this issue https://github.com/prettier/prettier-vscode/issues/3000?

**Comment by dangmai** [comment]
Hi,

I'm trying to understand the impact of this change for my plugin. Does this mean that I should ask all of my users to edit their Prettier config file to include something like:

```
{
  ... // rest of config
  "plugins": ["prettier-plugin-foo"]
}
```

or change their CLI call to include:

```
--plugin=prettier-plugin-foo
```

If that is the case, I can already foresee the amount of tickets opened because of this 😭 .

I haven't followed upstream development closely recently, my assumption is that this will only be in Prettier v3 and above, not v2 right?

**Comment by fisker** [comment]
Correct.

**Pull request merged**